File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletAirQuality.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletAirQuality;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Air Quality Bricklet

$ipcon = new IPConnection(); // Create IP connection
$aq = new BrickletAirQuality(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current all values
$all_values = $aq->getAllValues();

echo "IAQ Index: " . $all_values['iaq_index'] . "\n";

if ($all_values['iaq_index_accuracy'] == BrickletAirQuality::ACCURACY_UNRELIABLE) {
    echo "IAQ Index Accuracy: Unreliable\n";
} elseif ($all_values['iaq_index_accuracy'] == BrickletAirQuality::ACCURACY_LOW) {
    echo "IAQ Index Accuracy: Low\n";
} elseif ($all_values['iaq_index_accuracy'] == BrickletAirQuality::ACCURACY_MEDIUM) {
    echo "IAQ Index Accuracy: Medium\n";
} elseif ($all_values['iaq_index_accuracy'] == BrickletAirQuality::ACCURACY_HIGH) {
    echo "IAQ Index Accuracy: High\n";
}

echo "Temperature: " . $all_values['temperature']/100.0 . " °C\n";
echo "Humidity: " . $all_values['humidity']/100.0 . " %RH\n";
echo "Air Pressure: " . $all_values['air_pressure']/100.0 . " hPa\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Air Quality Bricklet

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletAirQuality aq = new BrickletAirQuality(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current all values
		int iaqIndex, temperature, humidity, airPressure; byte iaqIndexAccuracy;
		aq.GetAllValues(out iaqIndex, out iaqIndexAccuracy, out temperature, out humidity,
		                out airPressure);

		Console.WriteLine("IAQ Index: " + iaqIndex);

		if(iaqIndexAccuracy == BrickletAirQuality.ACCURACY_UNRELIABLE)
		{
			Console.WriteLine("IAQ Index Accuracy: Unreliable");
		}
		else if(iaqIndexAccuracy == BrickletAirQuality.ACCURACY_LOW)
		{
			Console.WriteLine("IAQ Index Accuracy: Low");
		}
		else if(iaqIndexAccuracy == BrickletAirQuality.ACCURACY_MEDIUM)
		{
			Console.WriteLine("IAQ Index Accuracy: Medium");
		}
		else if(iaqIndexAccuracy == BrickletAirQuality.ACCURACY_HIGH)
		{
			Console.WriteLine("IAQ Index Accuracy: High");
		}

		Console.WriteLine("Temperature: " + temperature/100.0 + " °C");
		Console.WriteLine("Humidity: " + humidity/100.0 + " %RH");
		Console.WriteLine("Air Pressure: " + airPressure/100.0 + " hPa");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
